Removed manual adding of SP & Facade

gr8shivam opened the pull request:
With Laravel 5.5 Pacakage AutoDiscovery, manually adding the Auth0 Service Provider and Facade is no more required.

Thus, Removed Auth0 Service Provider and Facade from `config/app.php` of both samples.

In accordance with https://github.com/auth0/docs/pull/6788

gr8shivam committed:
Update app.php

gr8shivam committed:
Update app.php

joshcanhelp reviewed (dismissed):
If I remove those two lines from `01-Login` and walk through the sample setup steps, I get this error:

```
Authentication user provider [auth0] is not defined. (View: /Users/josh-cunningham/Sites/laravel-webapp-sample/01-Login/resources/views/welcome.blade.php)
```

Was this change tested? If so, is there a step missing from the documentation?

Thanks!

gr8shivam commented:
Hi @joshcanhelp, it maybe because `auth0` package got installed earlier than `laravel` package. In my case, it's working perfectly. 

Since, it's uncertain, you can continue to use default method of adding provider in this sample. I'll thus close this pull request.

gr8shivam committed:
Removed auth0 from initial install

gr8shivam committed:
Added Auth0 installation doc

gr8shivam commented:
Removed `auth0/login` from `composer.json` and added Auth0 plugin installation doc in-line with Auth0 doc (https://auth0.com/docs/quickstart/webapp/laravel/01-login#install-the-auth0-plugin-and-its-dependencies).

gr8shivam commented:
Hi @joshcanhelp, try giving it a run now. 

Shouldn't fail since we'll be installing Auth0 plugin only after Laravel framework is installed. This also brings us in-line with the Auth0 Laravel documentation as @albertoperdomo suggested.

joshcanhelp commented:
Thanks @gr8shivam, I'll take a look as soon as I can.

joshcanhelp commented on 00-Starter-Seed/config/app.php:
```diff
@@ -162,7 +162,6 @@
         Illuminate\Translation\TranslationServiceProvider::class,
         Illuminate\Validation\ValidationServiceProvider::class,
         Illuminate\View\ViewServiceProvider::class,
-        Auth0\Login\LoginServiceProvider::class,
```
Let's leave this out since this sample is deprecated.

joshcanhelp replied:
Let's leave this out since this sample is deprecated.

joshcanhelp commented on 00-Starter-Seed/config/app.php:
```diff
@@ -225,7 +224,6 @@
         'URL' => Illuminate\Support\Facades\URL::class,
         'Validator' => Illuminate\Support\Facades\Validator::class,
         'View' => Illuminate\Support\Facades\View::class,
-        'Auth0' => \Auth0\Login\Facade\Auth0::class,
```
Let's leave this out since this sample is deprecated.

joshcanhelp replied:
Let's leave this out since this sample is deprecated.

joshcanhelp commented on 01-Login/README.md:
```diff
@@ -31,6 +31,15 @@ Otherwise:
 1. [Install Composer](https://getcomposer.org/doc/00-intro.md)
 1. [Run Composer](https://getcomposer.org/doc/01-basic-usage.md)
 
+### Installing Auth0 Plugin
```
We can leave this whole section out if you change the Composer command above this from:

```
composer install
```

... to:

```
composer require auth0/login:"~5.0"
```

This will install all the packages and do the auto-discovery.

gr8shivam replied:
Yup, we can go for it.

joshcanhelp reviewed (changes requested):
A few minor things. Thanks for sticking with us on this! I'll get this merged as soon as I hear back.

joshcanhelp commented:
@gr8shivam - Just wanted to check in one last time on this. Can you update this PR so we can get it merged in? Thank you!

joshcanhelp commented:
No activity
